pkg/daemon/update.go

```diff
@@ -566,12 +560,16 @@ func (dn *Daemon) writeUnits(units []ignv2_2types.Unit) error {
 			continue
 		}
 
-		// write the unit to disk
-		if err := writeFileAtomicallyWithDefaults(fpath, []byte(u.Contents)); err != nil {
-			return fmt.Errorf("failed to write systemd unit %q: %v", u.Name, err)
-		}
+		if u.Contents != "" {
+			glog.Infof("Writing systemd unit %q", u.Name)
+
+			// write the unit to disk
+			if err := writeFileAtomicallyWithDefaults(fpath, []byte(u.Contents)); err != nil {
+				return fmt.Errorf("failed to write systemd unit %q: %v", u.Name, err)
+			}
 
-		glog.V(2).Infof("Successfully wrote systemd unit %q: ", u.Name)
+			glog.V(2).Infof("Successfully wrote systemd unit %q: ", u.Name)
+		}
```

Comment:
I believe the change in behavior here is that, after this lands, if someone tries to enable a non-existant systemd unit, this is now going to crashloop.

Reply:
I thought it would just issue a warning. Let me try that out and arrange a fix for that issue.

Reply:
I believe this is fine tho, it's what systemctl does when the unit doesn't exist actually. Let me think one more minute